plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/animators/PuckAnimatorManager.kt

```diff
@@ -62,16 +62,7 @@ internal class PuckAnimatorManager(
     vararg targets: Double,
     options: (ValueAnimator.() -> Unit)?
   ) {
-    val optimized = DoubleArray(targets.size)
-    targets.toTypedArray().apply {
-      for (i in 0 until size) {
-        optimized[i] = if (i == 0)
-          MathUtils.normalize(get(i))
-        else
-          MathUtils.shortestRotation(MathUtils.normalize(get(i)), optimized[i - 1])
-      }
-    }
-    bearingAnimator.animate(*optimized.toTypedArray(), options = options)
+    bearingAnimator.animate(*MathUtils.prepareOptimalBearingPath(targets).toTypedArray(), options = options)
```

Comment:
I don't think we want to optimize the path for the puck. If a turn is tighter than 180 degrees, we still want to follow the true rotation, not use the shortest path.

Reply:
Capturing a chat we had with @kiryldz - in the end, the baked-in shortest path is probably a safer assumption to have, it will work okay 99% of the time without developers having to think. We can always expose a `shortestPath` flag in the future if this becomes a request.